### ISSUE TITLE
Kernel build: arch/x86/tools/insn_decoder_test: error: malformed line

### ISSUE THREAD

**Issue by dhodovsk** [Rust-for-Linux/linux#1134]
While testing the mainline kernel, we get an error during the x86_64 build:
```
00:29:20 arch/x86/tools/insn_decoder_test: error: malformed line 3733181:
00:29:20 2_>:ffffffff81c92050
00:29:20 make[4]: *** [arch/x86/tools/Makefile:26: posttest] Error 3
00:29:20 make[3]: *** [arch/x86/Makefile:310: bzImage] Error 2
00:29:20 make[3]: *** Waiting for unfinished jobs....
```

For us, it started happening two weeks ago, since the [Merge tag 'rust-6.13' of https://github.com/Rust-for-Linux/linux](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=798bb342e0416d846cf67f4725a3428f39bfb96b) commit in kernel mainline.

The issue was already discussed in [linux-kernel Mailing List](https://lore.kernel.org/lkml/Y9ES4UKl%2F+DtvAVS@gmail.com/T/), but it seems the suggested patches didn't come through.

Links:
- list of the affected trees and branches: https://datawarehouse.cki-project.org/issue/3332 
- our pipeline: https://gitlab.com/redhat/red-hat-ci-tools/kernel/cki-internal-pipelines/cki-trusted-contributors/-/jobs/8607023441
- the full build log: https://s3.amazonaws.com/arr-cki-prod-trusted-artifacts/trusted-artifacts/1583189702/build_x86_64/8607023441/artifacts/build.log
- kernel config: https://s3.amazonaws.com/arr-cki-prod-trusted-artifacts/trusted-artifacts/1583189702/build_x86_64/8607023441/artifacts/kernel-upstream-mainline-redhat_1583189702-x86_64-kernel.config

---
I first wrongly [reported this in the clang](https://github.com/ClangBuiltLinux/linux/issues/2060) repo and they pointed me to this repository and also provided a reproducer :)
```$ git show -s --format='%h ("%s")'
f92f4749861b ("Merge tag 'clk-fixes-for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/clk/linux")

$ rustc --version
rustc 1.83.0 (90b35a623 2024-11-26)

$ clang --version | head -1
ClangBuiltLinux clang version 19.1.5 (https://github.com/llvm/llvm-project.git ab4b5a2db582958af1ee308a790cfdb42bd24720)

$ curl -LSso .config https://s3.amazonaws.com/arr-cki-prod-trusted-artifacts/trusted-artifacts/1583189702/build_x86_64/8607023441/artifacts/kernel-upstream-mainline-redhat_1583189702-x86_64-kernel.config

$ make -skj"$(nproc)" ARCH=x86_64 LLVM=1 olddefconfig bzImage
...
arch/x86/tools/insn_decoder_test: error: malformed line 3735255:
2_>:ffffffff81c94190
make[4]: *** [arch/x86/tools/Makefile:26: posttest] Error 3
...
```

**Comment by ojeda** [comment]
Thanks for the report -- yeah, the issue is known and old. The latest report (and patch) about it that I am aware of is https://lore.kernel.org/rust-for-linux/CANiq72nyMi5NUwmg20ZzX8RNC8+dP2J3Zey9Qfn+5CaoeET+7A@mail.gmail.com/.

Could you please reply (ping) in Lore, i.e. in one of the proposed patches like the one above? If this is breaking a particularly important use case (e.g. at your company), then it may be useful to mention that clearly in the message so that they raise the priority of this.

Thanks!

**Comment by sergiocollado** [comment]
Hello, 
 This should be fixed with this patch: 
 https://lore.kernel.org/rust-for-linux/CABVgOSm=5Q0fM6neBhxSbOUHBgNzmwf2V22vsYC10YRBT=kN1g@mail.gmail.com/T/#t

**Comment by ojeda** [comment]
Yeah, that is commit c104c16073b7 ("Kunit to check the longest symbol length") that landed in v6.15, so I think this can be closed.

Having said that, should it be backported to the LTSs? i.e. at least the `arch/x86/tools/insn_decoder_test.c` part. (I replied on the list since I cannot find the handles of everybody, sorry! @sulix @shuahkh)

If so, @sergiocollado, would you like to submit it? It is not a complex procedure, and may be a nice experience (https://docs.kernel.org/process/stable-kernel-rules.html). Otherwise, I or someone else can do that.

Thanks!

**Comment by sergiocollado** [comment]
sure, I will backport it
